site/app/templates/course/CourseMaterials.twig

```diff
@@ -155,8 +155,47 @@
 
 {% macro display_folder(name, course_materials, links, user_group, user_section, id, indent, display_file_url, folder_path, directory_priorities, material_list, date_format, folder_ids, seen) %}
     {% set showFolder = false %}
-    {% import _self as macros %}
-    {{ macros.hideFolderCheck(course_materials,indent,user_group,user_section)}}
+    {% for course_material in course_materials %}
```

Comment:
The indentation here is way too deep.  Some of this logic should either be moved to a macro (like it was originally), or simplified to have less indentation.  In addition, the logic is repetitive and doesn't handle cases beyond 5 levels of nestedness.

Reply:
I didn't find a way to create a recursive macro, since macro can't return anything and the result of checking hidden material will lost after every loop. I also not sure how to simplify this.